File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map <int,int> sinai;
        for(int i=0;i<nums.size();i++){
            int comp=target-nums[i];
            if (sinai.find(comp)!=sinai.end()){
                return {sinai[comp],i};
            }
            sinai[nums[i]]=i;

            }
    
    return {};
    }};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        b={}
        for i in range(len(nums)):
            comp = target - nums[i]
            if comp in b:
                return [i,b[comp]]
            b[nums[i]]=i
        return -1